grype doesn't work with Go 1.14

**What happened**: Tried using grype with Go 1.14 which lead to errors:

```
Error: ../../../../go/pkg/mod/github.com/anchore/syft@v0.14.1-0.20210405182557-7a10cbae0c41/syft/file/digest_cataloger.go:90:38: hash.String undefined (type crypto.Hash has no field or method String)
Error: ../../../../go/pkg/mod/github.com/anchore/syft@v0.14.1-0.20210405182557-7a10cbae0c41/syft/file/secrets_search_by_line_strategy.go:99:18: pattern.SubexpIndex undefined (type *regexp.Regexp has no field or method SubexpIndex)
Error: Process completed with exit code 2.
```

**What you expected to happen**: At the very least, something in the documentation that mentions what are the Go versions that are supported. Ideally (unsure if this is possible) something that prevents usage in unsupported Go versions, or that produces a better error

**How to reproduce it (as minimally and precisely as possible)**: Run grype with Go 1.14

--- a/README.md
+++ b/README.md
@@ -5,6 +5,7 @@
 [![Go Report Card](https://goreportcard.com/badge/github.com/anchore/grype)](https://goreportcard.com/report/github.com/anchore/grype)
 [![GitHub release](https://img.shields.io/github/release/anchore/grype.svg)](https://github.com/anchore/grype/releases/latest)
 [![License: Apache-2.0](https://img.shields.io/badge/License-Apache%202.0-blue.svg)](https://github.com/anchore/grype/blob/main/LICENSE)
+[![GitHub go.mod Go version](https://img.shields.io/github/go-mod/go-version/anchore/grype.svg)](https://github.com/anchore/grype)
 
 A vulnerability scanner for container images and filesystems. [Easily install the binary](#installation) to try it out.
 

--- a/go.mod
+++ b/go.mod
@@ -1,6 +1,6 @@
 module github.com/anchore/grype
 
-go 1.14
+go 1.16
 
 require (
 	github.com/acarl005/stripansi v0.0.0-20180116102854-5a71ef0e047d

